pkg/gitprovider/git_provider.go

```diff
@@ -14,15 +14,16 @@ import (
 const personalNamespaceId = "<PERSONAL>"
 
 type StaticGitContext struct {
-	Id       string  `json:"id"`
-	Url      string  `json:"url"`
-	Name     string  `json:"name"`
-	Branch   *string `json:"branch,omitempty"`
-	Sha      *string `json:"sha,omitempty"`
-	Owner    string  `json:"owner"`
-	PrNumber *uint32 `json:"prNumber,omitempty"`
-	Source   string  `json:"source"`
-	Path     *string `json:"path,omitempty"`
+	Id         string  `json:"id"`
+	Url        string  `json:"url"`
+	Name       string  `json:"name"`
+	Branch     *string `json:"branch,omitempty"`
+	Sha        *string `json:"sha,omitempty"`
+	Owner      string  `json:"owner"`
+	PrNumber   *uint32 `json:"prNumber,omitempty"`
+	Source     string  `json:"source"`
+	Path       *string `json:"path,omitempty"`
+	ProjectKey string  `json:"projectKey,omitempty"`
```

Comment:
Why is the `ProjectKey` needed? Isn't the `Id` enough?

Reply:
The getcommit api requires args as project key and repository id/name/slug ..

`func (a *DefaultApiService) GetCommits(project, repository string, localVarOptionals map[string]interface{}) (*APIResponse, error) {`

Reply:
Then store the project key in the repo Id. It doesn't make sense to introduce a property specific to the bitbucket server provider.

Reply:
Also, please do not mark conversations as resolved yourself. The reviewer will mark them once they confirm the resolution. Thanks.

Reply:
done